src/data/equiv/module.lean

```diff
@@ -235,6 +235,56 @@ lemma symm_apply_eq {x y} : e.symm x = y ↔ x = e y := e.to_equiv.symm_apply_eq
 lemma eq_symm_apply {x y} : y = e.symm x ↔ e y = x := e.to_equiv.eq_symm_apply
 omit σ'
 
+lemma eq_comp_symm (f : M₂ → M₃) (g : M₁ → M₃) :
+  f = g ∘ e₁₂.symm ↔ f ∘ e₁₂ = g := e₁₂.to_equiv.eq_comp_symm f g
+
+lemma comp_symm_eq (f : M₂ → M₃) (g : M₁ → M₃) :
+  g ∘ e₁₂.symm = f ↔ g = f ∘ e₁₂ := e₁₂.to_equiv.comp_symm_eq f g
+
+lemma eq_symm_comp (f : M₃ → M₁) (g : M₃ → M₂) :
+  f = e₁₂.symm ∘ g ↔ e₁₂ ∘ f = g := e₁₂.to_equiv.eq_symm_comp f g
+
+lemma symm_comp_eq (f : M₃ → M₁) (g : M₃ → M₂) :
+  e₁₂.symm ∘ g = f ↔ g = e₁₂ ∘ f := e₁₂.to_equiv.symm_comp_eq f g
+
+variables [ring_hom_comp_triple σ₂₁ σ₁₃ σ₂₃] [ring_hom_comp_triple σ₃₁ σ₁₂ σ₃₂]
+
+include module_M₃
+
+lemma eq_comp_to_linear_map_symm (f : M₂ →ₛₗ[σ₂₃] M₃) (g : M₁ →ₛₗ[σ₁₃] M₃) :
+  f = g.comp e₁₂.symm.to_linear_map ↔ f.comp e₁₂.to_linear_map = g :=
+begin
+  split; intro H; ext,
+  simp [H, e₁₂.to_equiv.eq_comp_symm f g],
+  simp [←H, ←e₁₂.to_equiv.eq_comp_symm f g],
```

Comment:
```suggestion
  { simp [H, e₁₂.to_equiv.eq_comp_symm f g] },
  { simp [←H, ←e₁₂.to_equiv.eq_comp_symm f g] }
```
and throughout, is the mathlib style; it ensures that if something breaks in the first `simp` it doesn't just run the second one on the first goal, which would be confusing.